crypto/s2n_rsa_pss.h

```diff
@@ -17,37 +17,28 @@
 
 #include <stdint.h>
 #include <s2n.h>
-#include <openssl/evp.h>
 
-#include "crypto/s2n_hash.h"
-#include "utils/s2n_blob.h"
+#include "crypto/s2n_openssl.h"
+#include "crypto/s2n_rsa.h"
+
+#define RSA_PSS_SIGN_VERIFY_RANDOM_BLOB_SIZE    32
+#define RSA_PSS_SIGN_VERIFY_SIGNATURE_SIZE      256
+
+#ifndef EVP_PKEY_RSA_PSS
+#define EVP_PKEY_RSA_PSS EVP_PKEY_NONE
+#endif
 
 /* OpenSSL 1.1.1d  10 Sep 2019 is broken, so disable on that version. For further info see: crypto/evp/p_lib.c:469
  *
  * This feature requires this Openssl commit for Openssl 1.1.x versions: openssl/openssl@4088b92
  */
 #if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 1) && !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER != 0x1010104fL
```

Comment:
now that we are doing RSA PSS signing correctly - PSS signing should be available in openssl 1.0.2, and PSS certs OPENSSL_VERSION_NUMBER > 0x1010104fL. would it be possible to make this distinction here?

Reply:
The RSA-PSS signing code as-is will not compile with Openssl 1.0.2. If you need RSA-PSS pre-1.0.2, open an issue and we can consider rewriting our RSA-PSS code without any of the newer functions.